### PR TITLE
Use continuous hash-table initialization for BaseLineHashJoin

### PR DESCRIPTION
**Pull request by dchigarev**
Continuous of #208 ([comment](https://github.com/intel-ai/omniscidb/pull/208#issuecomment-975387265))

Signed-off-by: Dmitry Chigarev <dmitry.chigarev@intel.com>

### CODE REVIEW

**Comment by dchigarev** [comment]
@ienkovich @fexolm Requesting your review.
(It seems that I have limited access to this repo and can't explicitly request a review, so I'm just pinging you)

File: QueryEngine/JoinHashTable/Runtime/HashJoinRuntime.cpp [hunk 1]
```diff
@@ -357,15 +357,18 @@ DEVICE void SUFFIX(init_baseline_hash_join_buff)(int8_t* hash_buff,
...
-  int32_t step = blockDim.x * gridDim.x;
+  const int32_t start = threadIdx.x + blockDim.x * blockIdx.x;
+  const int64_t stop = entry_count;
+  const int32_t step = blockDim.x * gridDim.x;
 #else
-  int32_t start = cpu_thread_idx;
-  int32_t step = cpu_thread_count;
+  const int64_t chunk_size = entry_count / cpu_thread_count;
+  const int64_t start = cpu_thread_idx * chunk_size;
+  const int64_t stop = (cpu_thread_idx == cpu_thread_count - 1) ? entry_count : (cpu_thread_idx + 1) * chunk_size;
```
Comment (ienkovich) [inline QueryEngine/JoinHashTable/Runtime/HashJoinRuntime.cpp:366]: `end` is a commonly used name for such cases. More even range split:
```
start = cpu_thread_idx * entry_count / cpu_thread_count;
end = (cpu_thread_idx + 1) * entry_count / cpu_thread_count;
```

File: QueryEngine/JoinHashTable/Runtime/HashJoinRuntime.cpp [hunk 2]
```diff
@@ -357,15 +357,18 @@ DEVICE void SUFFIX(init_baseline_hash_join_buff)(int8_t* hash_buff,
                                                  const int32_t cpu_thread_idx,
                                                  const int32_t cpu_thread_count) {
 #ifdef __CUDACC__
-  int32_t start = threadIdx.x + blockDim.x * blockIdx.x;
-  int32_t step = blockDim.x * gridDim.x;
+  const int32_t start = threadIdx.x + blockDim.x * blockIdx.x;
+  const int64_t stop = entry_count;
```
Comment (ienkovich) [inline QueryEngine/JoinHashTable/Runtime/HashJoinRuntime.cpp:361]: Please stick to a single type for these.

Comment (dchigarev) [reply QueryEngine/JoinHashTable/Runtime/HashJoinRuntime.cpp]: now it uses `int64_t` everywhere here

File: QueryEngine/JoinHashTable/Runtime/HashJoinRuntime.cpp (hunk 1 above)
Comment (dchigarev) [reply QueryEngine/JoinHashTable/Runtime/HashJoinRuntime.cpp]: done

**Review by ienkovich — APPROVED** [review]
LGTM, thanks!

**Pull request merged**